components/infobox/commons/infobox_person.lua

```diff
@@ -24,187 +23,211 @@ local function GetBirthAndDeath()
+			}
+		},
+		Cell{name = 'Clan', content = {
+				self:_createTeam(args.clan, args.clanlink),
+				self:_createTeam(args.clan2, args.clanlink2)
+			}
+		},
+		Cell{name = 'Alternate IDs', content = {args.ids or args.alternateids}},
+		Cell{name = 'Nicknames', content = {args.nicknames}},
+		Builder{
+			builder = function()
+				if earnings and earnings ~= 0 then
+					return {
+						Cell{name = 'Total Earnings', content = {'$' .. Language:formatNum(earnings)}},
+					}
+				end
+			end
+		},
+		Customizable{id = 'custom', children = {}},
+		Builder{
+			builder = function()
+				if not Table.isEmpty(links) then
+					return {
+						Title{name = 'Links'},
+						Widgets.Links{content = links, variant = _LINK_VARIANT}
+					}
+				end
+			end
+		},
+		Customizable{id = 'achievements', children = {
+				Builder{
+					builder = function()
+						if not String.isEmpty(args.achievements) then
+							return {
+								Title{name = 'Achievements'},
+								Center{content = {args.achievements}}
+							}
+						end
+					end
+				},
+			}
+		},
+		Customizable{id = 'history', children = {
+				Builder{
+					builder = function()
+						if not String.isEmpty(args.history) then
+							return {
+								Title{name = 'History'},
+								Center{content = {args.history}}
+							}
+						end
+					end
+				},
+			}
+		},
+		Center{content = {args.footnotes}},
+		Customizable{id = 'customcontent', children = {}},
+	}
+
+	infobox:bottom(self:createBottomContent())
+
+	local statusToStore
+	if _should_store_data then
+		Variables.varDefine('earnings', earnings)
+		statusToStore = self:getStatusToStore(args)
+		infobox:categories(unpack(self:getCategories(
+					args,
+					birthDisplay,
+					personType.category,
+					statusToStore
+				)))
 	end
 
-	return infobox:build()
-end
-
---- Allows for overriding this functionality
-function Person:getCategories(infobox, args, role, status)
-	infobox:categories(role .. 's')
-	if not args.teamlink and not args.team then
-		infobox:categories('Teamless ' .. role .. 's')
-	end
-	if args.country2 or args.nationality2 then
-		infobox:categories('Dual Citizenship ' .. role .. 's')
-	end
-	if args.death_date then
-		infobox:categories('Deceased ' .. role .. 's')
-	end
-	if
-		args.retired == 'yes' or args.retired == 'true'
-		or string.lower(status or '') == 'retired'
-		or string.match(args.retired or '', '%d%d%d%d%')--if retired has year set apply the retired category
-	then
-		infobox:categories('Retired ' .. role .. 's')
-	else
-		infobox:categories('Active ' .. role .. 's')
-	end
-	if not args.id then
-		infobox:categories('InfoboxIncomplete.')
-	end
-	if not args.image then
-		infobox:categories(role .. 's with no profile picture')
-	end
-	if not args.birthDisplay then
-		infobox:categories(role .. 's with unknown birth date')
+	local builtInfobox = infobox:widgetInjector(self:createWidgetInjector()):build(widgets)
+
+	if _should_store_data then
+		self:_setLpdbData(
```

Comment:
I really dislike this flow with duplicated ifs. Can this not be avoided at all?

Reply:
problem is that the categories have to be set before we build it, right?

an alternative could be something like:
```lua
	if _shouldStoreData then
		Variables.varDefine('earnings', earnings)
		local statusToStore = self:getStatusToStore(args)
		infobox:categories(unpack(self:getCategories(
					args,
					birthDisplay,
					personType.category,
					statusToStore
				)))

		local builtInfobox = infobox:widgetInjector(self:createWidgetInjector()):build(widgets)

		self:_setLpdbData(
			args,
			links,
			birthday,
			deathday,
			statusToStore,
			personType.store,
			earnings
		)

		return builtInfobox
	end

	return infobox:widgetInjector(self:createWidgetInjector()):build(widgets)
```

Reply:
Can we not just always set vars and categories and then just have LPDB inside the if?

Reply:
if we want to have the categories include non main space, sure

(the var is irrelevant imo (only needed for smw storage))

Reply:
RL does that, don't know about other wikis though

Reply:
how about moving the `if _shouldStoreData then` for the categories into the customizable function?
the flow of code would be better in the `:createInfobox()` function and would allow the wikis to decide themselves if they want to enable the categories in non main space

Reply:
Sure, let's do that!